website/blog/2025-06-05-replacing-mongodb-with-ferretdb-librechat.md

```diff
@@ -0,0 +1,263 @@
+---
+slug: replacing-mongodb-with-ferretdb-librechat
+title: 'Replacing MongoDB with FerretDB in LibreChat'
+authors: [alex]
+description: >
+  A guide to replacing MongoDB with FerretDB for a fully open-source LibreChat setup that runs the Ollama phi4-mini model locally.
+image: /img/blog/ferretdb-librechat.jpg
+tags:
+  [
+    tutorial,
+    document databases,
+    community,
+    open source,
+    compatible applications
+  ]
+---
+
+![Replacing MongoDB with FerretDB on LibreChat](/img/blog/ferretdb-librechat.jpg)
+
+[LibreChat](https://www.librechat.ai/) is a free, open-source application that provides a user-friendly and customizable interface for interacting with various AI providers and models.
+
+<!--truncate-->
+
+It allows users to connect with cloud providers like [OpenAI](https://openai.com/), [Azure](https://azure.microsoft.com/), [Anthropic](https://www.anthropic.com/), and others as well as fully open-source tools like [Ollama](https://ollama.com/), which lets you run models like `phi4-mini` locally.
+
+For LibreChat users who want to stay fully open source, FerretDB is a great drop-in replacement for MongoDB,
+especially if you're looking to avoid proprietary databases or vendor lock-in.
+It uses PostgreSQL with DocumentDB extension as the backend, while letting you use familiar MongoDB operations and commands.
+
+This guide shows how to run LibreChat with FerretDB (instead of MongoDB) and Ollama to run the `phi4-mini` model locally – with all components being fully open source.
+
+## Prerequisites
+
+To follow this guide, ensure:
+
+- [Docker](https://www.docker.com/) is installed and running on your machine.
+- Ollama is installed and running on your machine.
+  If you haven't installed Ollama yet, download it from the [official website](https://ollama.com/download).
+  This guide uses the `phi4-mini` model, an open-source model optimized for reasoning and accuracy in text-based tasks.
+  Once Ollama is running, pull the `phi4-mini` model:
+
+  ```sh
+  ollama pull phi4-mini
+  ```
+
+## How to set up LibreChat with FerretDB
+
+Start by cloning the LibreChat repository:
+
+```sh
+git clone https://github.com/danny-avila/LibreChat.git
+cd LibreChat
+```
+
+You can find more instructions on how to set up LibreChat in the [LibreChat documentation](https://www.librechat.ai/docs/quick_start/local_setup).
+
+Next, copy the `.env.example` file to `.env`:
+
+```sh
+cp .env.example .env
+```
+
+You may need to adjust the environment variables in `.env` to suit your setup.
+For instance, you may have set the `UID` and `GID` to `999` in the `.env` file, as it was the case in this guide.
```

Comment:
instead of saying _as it was the case in this guide._, could you say _if you get xyz error_? Also it gives no information why it was done, could you add that please.